lib/models/authentication/authentication_service.dart

```diff
@@ -176,9 +176,14 @@ class AuthenticationService {
       // not match the nonce in `appleCredential.identityToken`, sign in will fail.
       final UserCredential userCredential =
           await _firebaseAuth.signInWithCredential(oauthCredential);
-
-      final User? user = userCredential.user;
-
+      User? user = userCredential.user;
+      // TODO: Handle when user hide email when sign in with apple id
+      if (user != null) {
```

Comment:
should add a check here to only assign user's display name if user does not have a display name